source/module_base/global_file.cpp

```diff
@@ -122,6 +124,45 @@ void ModuleBase::Global_File::make_dir_out(
 #endif
     }
 
+    // make dir for HS matrix output in md calculation
+    if(out_hs && (calculation == "md" || calculation == "sto-md"))
+    {
+        int make_dir_matrix = 0;
+        std::string command1 =  "test -d " + GlobalV::global_matrix_dir + " || mkdir " + GlobalV::global_matrix_dir;
+
+        times = 0;
+        while(times<GlobalV::NPROC)
```

Comment:
It seems here you are trying to make the directory from each process? Is there any specific reason to do it this way?

Reply:
I imitate the way similar to the main directory,